public/coffee/ide/review-panel/controllers/ReviewPanelController.coffee

```diff
@@ -27,6 +28,16 @@ define [
 			rendererData: {}
 			loadingThreads: false
 
+		$scope.$on "project:joined", () ->
+			$scope.reviewPanel.overview.docsCollapsedState = JSON.parse(localStorage("docs_collapsed_state:#{$scope.project_id}")) or {}
+
+		window.addEventListener "beforeunload", () ->
+			collapsedStates = {}
+			for doc, state of $scope.reviewPanel.overview.docsCollapsedState
+				collapsedStates[doc] = state if state is true
```

Comment:
I find this inline syntax hard to parse and never trust it. Maybe make it explicit?